llvm/test/CodeGen/AMDGPU/si-lower-sgpr-spills-vgpr-lanes-usage.mir

```diff
@@ -0,0 +1,231 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 5
+# RUN: llc -mtriple=amdgcn-amd-amdhsa -mcpu=gfx908 -verify-machineinstrs -stress-regalloc=3 -start-before=greedy -stop-after=si-lower-sgpr-spills -o - %s | FileCheck -check-prefix=SGPR_SPILLED %s
```

Comment:
This test would be more reliable if it did not depend on register allocation behavior. Can you only run the post-RA passes?

Reply:
I did not understand much what you asked me to do, Can you elaborate more?

Reply:
Change the RUN line to use --start-before=stack-slot-coloring --stop-after=si-lower-sgpr-spills. The original test you have written should be modified first. To use this new RUN line, the test should be the output of sgpr-regalloc pipeline just before invoking the newly introduced stack slot coloring.

Reply:
Now, understood, thanks for the clarification!

Reply:
As I was trying to generate post optimized SGPR spills by adding " --start-before=stack-slot-coloring --stop-after=si-lower-sgpr-spills", I found that if stackSlotColoring(SSC) is introduced before SILowerSGPRSpills, it really depends on regAlloc. As StackSlotColoring pass uses LiveStack(LS) analysis results {which is currently computed at RegAlloc phase, as LS itself do not do anything}. So if we invoke SSC directly on 'input.mir' without LS results available, we acutually won't see any stack slots optimization (in turn no sgpr spill optimization).

Reply:
So, I guess this test file need to consider to start from RA rather than post RA else, SCC really won't work thereafter(with optimized SGPR spill code) because of absence of LS , which really gets calculated at RA.

Reply:
The pipeline is stopped way before VGPR allocation here, as it happens much after SILowerSGPR spills. The -start-before=greedy, here it refers to SGPR greedy allocation, followed by virtRegRewriter & SILOwerSGPRSpill. I feel the change needed would eith to use -sgpr-regalloc option OR use  -start-before=greedy,0 to depict using SGPR allocation.

Reply:
Just like to add, VGPR allocation is happening after SGPR allocation already happened.

Reply:
@cdevadas, can you add your thoughts to it!

Reply:
I think it is reasonable to start the test from sgpr-regalloc and keep it until sgpr spill lowering to capture how the newly inserted stack slot coloring helps optimize vgpr lanes. I don't see any problem with the test.